src/main/java/uk/ac/ebi/eva/pipeline/io/readers/NonAnnotatedVariantsMongoReader.java

```diff
@@ -31,39 +41,73 @@
  * {@link org.springframework.batch.item.data.MongoItemReader} is using
  * pagination and it is slow with large collections
  */
-public class NonAnnotatedVariantsMongoReader extends MongoDbCursorItemReader {
+public class NonAnnotatedVariantsMongoReader implements ItemStreamReader<VariantWrapper>, InitializingBean {
+
+    private MongoDbCursorItemReader delegateReader;
+
+    private DBObjectToVariantConverter converter;
 
     private static final String STUDY_KEY = VariantToDBObjectConverter.FILES_FIELD + "."
             + VariantSourceEntryToDBObjectConverter.STUDYID_FIELD;
 
     /**
-     * @param studyId Can be the empty string, meaning to bring all non-annotated variants in the collection.
-     *  If the studyId string is not empty, bring only non-annotated variants from that study. This parameter should
-     *  not be null in any case.
+     * @param studyId Can be the empty string, meaning to bring all non-annotated variants in the collection. If the
+     * studyId string is not empty, bring only non-annotated variants from that study. This parameter should not be null
+     * in any case.
      */
     public NonAnnotatedVariantsMongoReader(MongoOperations template, String collectionsVariantsName, String studyId) {
         if (studyId == null) {
             throw new IllegalArgumentException("NonAnnotatedVariantsMongoReader needs a non-null studyId " +
                     "(it can take a studyId or an empty string for reading every study)");
```

Comment:
I know we discussed it #100 , but I'm still not convinced about this interface.

What do you think of allowing the study ID to be null, and check that along with being empty before adding it as a condition? That would ensure that all cases of "unspecified study" behave the same way.